### DISCUSSION TITLE
oAuth2 Provider Application

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#9431] [Archive]
Has someone a working oAuth2 Provider, which can be used with directus. I mean not an already existing one, but one which i could host.
I have tried some but it hangs at the /token call, which then won't finish.

**Comment by rijkvanzanten** [comment]
You can self-host [Keycloak](https://www.keycloak.org) pretty easily through Docker; that's what we use for debugging as well:

https://github.com/directus/directus/blob/3ca490289ae7c08ac9db3868470acfc61bdfd902/docker-compose.yml#L134-L140
